include/grpc/impl/codegen/port_platform.h

```diff
@@ -127,6 +127,9 @@
 #define GPR_PLATFORM_STRING "android"
 #define GPR_ANDROID 1
 #ifdef __ANDROID_API__
```

Comment:
Although `__ANDROID_API__` should always be defined, we probably want to also emit an error if it is not defined, just to be safe